ddcz/views.py

```diff
@@ -599,3 +602,28 @@ def tavern(request):
             "current_list_style": list_style,
         },
     )
+
+
+def sign_up(request):
+    form = SignUpForm()
+
+    if request.method == "POST" and request.POST["submit"]:
+        data = SignUpForm.set_for_save(request.POST.copy())
```

Comment:
So this shouldn't be needed, the pattern should be just `form = SignUpForm(request.POST)`, see [how Django does it in the tutorial](https://docs.djangoproject.com/en/3.1/topics/forms/#the-view). 

I see what you are trying to do and I think the correct way is to override the `clean()` method [of the model](https://docs.djangoproject.com/en/3.2/ref/models/instances/#django.db.models.Model.clean).